src/main.js

```diff
@@ -90,8 +93,39 @@ function compileTo(src, dest) {
     }
 }
 
+function compileAndRun(src) {
+    var js = compileSource(src);
+    var script = new vm.Script(js);
+    var root = global;
+    root.require = function(id) {
+        try {
+            return require(path.resolve(id));
+        } catch (Error) {
+            return require(id);
+        }
+    };
+    root.__filename = path.normalize(path.join(process.cwd(), src));
+    root.__dirname = path.dirname(root.__filename) + "/";
```

Comment:
I don't think this needs the `/` at the end. If I do `node -p "__dirname"` it simply prints out `.`.